File: DeliveryManagementAPI/Models/LocationCheckpoint.cs
namespace DeliveryManagementAPI.Models
{
    /// <summary>
    /// Điểm check in vị trí trong quá trình giao hàng
    /// </summary>
    public class LocationCheckpoint
    {
        [System.ComponentModel.DataAnnotations.Key]
        public int CheckpointId { get; set; }
        public int OrderId { get; set; }
        public double Latitude { get; set; }  // Vĩ độ
        public double Longitude { get; set; } // Kinh độ
        public string LocationName { get; set; } = string.Empty;
        public DateTime CheckInTime { get; set; }
        public string Notes { get; set; } = string.Empty;
    }
}

File: ShipperAndroidApp/app/src/main/java/com/example/shipperapp/models/LocationCheckpoint.java
package com.example.shipperapp.models;

import com.google.gson.annotations.SerializedName;

public class LocationCheckpoint {
    @SerializedName("checkpointId")
    public int checkpointId;
    
    @SerializedName("orderId")
    public int orderId;
    
    @SerializedName("latitude")
    public double latitude;
    
    @SerializedName("longitude")
    public double longitude;
    
    @SerializedName("locationName")
    public String locationName;
    
    // checkInTime: Backend tự set, không gửi từ client
    @SerializedName("checkInTime")
    public String checkInTime; // Chỉ dùng khi nhận response
    
    @SerializedName("notes")
    public String notes;
    
    // Constructor for sending check-in
    public LocationCheckpoint() {
        this.locationName = "";
        this.notes = "";
    }
}
